src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java

```diff
@@ -0,0 +1,279 @@
+                .defendantResponseDocuments(mapSystemGeneratedcaseDocument(caseData.getDefendantResponseDocuments(),
+                                                                           null))
+                .applicant1(caseData.getApplicant1())
+                .respondent1(caseData.getRespondent1())
+                .courtLocation(caseData.getHearingLocation().getValue().getLabel())
+                .hearingDate(null != caseData.getHearingDate()
+                                 ? DateFormatHelper.formatLocalDate(caseData.getHearingDate(), "dd-MM-yyyy") : null)
+                .ccdCaseReference(caseData.getCcdCaseReference())
+                .build();
+        bundlingCaseData = mapRespondent2Applicant2Details(bundlingCaseData, caseData);
+        return  bundlingCaseData;
+    }
+
+    private BundlingCaseData mapRespondent2Applicant2Details(BundlingCaseData bundlingCaseData, CaseData caseData) {
+        if (null != caseData.getAddApplicant2() && caseData.getAddApplicant2().equals(YesOrNo.YES)) {
+            bundlingCaseData.toBuilder().hasApplicant2(true);
+        }
+        if (null != caseData.getAddRespondent2() && caseData.getAddRespondent2().equals(YesOrNo.YES)) {
+            bundlingCaseData.toBuilder().hasRespondant2(true);
+        }
+        if (null != caseData.getApplicant2()) {
+            bundlingCaseData.toBuilder().applicant2(caseData.getApplicant2());
+        }
+        if (null != caseData.getRespondent2()) {
+            bundlingCaseData.toBuilder().respondent2(caseData.getRespondent2());
+        }
+        return bundlingCaseData;
+    }
+
+    private ServedDocument mapServedDocuments(ServedDocumentFiles servedDocumentFiles) {
+        List<BundlingRequestDocument> bundlingServedDocFiles = new ArrayList<>();
+        if (!Optional.ofNullable(servedDocumentFiles).isEmpty() && null != servedDocumentFiles.getParticularsOfClaimDocument()) {
+            servedDocumentFiles.getParticularsOfClaimDocument().forEach(document -> {
+                bundlingServedDocFiles.add(BundlingRequestDocument.builder()
+                                               .documentFileName(document.getValue().getDocumentFileName())
+                                               .documentLink(DocumentLink.builder()
+                                                                 .documentUrl(document.getValue().getDocumentUrl())
+                                                                 .documentBinaryUrl(document.getValue().getDocumentBinaryUrl())
+                                                                 .documentFilename(document.getValue().getDocumentFileName()).build())
+                                               .build());
+
+            });
+        }
+        List<Element<BundlingRequestDocument>> particulars = ElementUtils.wrapElements(bundlingServedDocFiles);
+        return  ServedDocument.builder().particularsOfClaimDocument(particulars).build();
+    }
```

Comment:
Try short-circuiting the "if" statement in this function.

With that done, you'll notice that ultimately you just want to process things if they exist or work against an empty collection otherwise.

It may lead you to the following:
```
    private ServedDocument mapServedDocuments(ServedDocumentFiles servedDocumentFiles) {
        List<BundlingRequestDocument> bundlingServedDocFiles = Optional.of(servedDocumentFiles)
            .map(ServedDocumentFiles::getParticularsOfClaimDocument)
            .orElse(new ArrayList<>())
            .stream()
            .map(document -> BundlingRequestDocument.builder()
                .documentFileName(document.getValue().getDocumentFileName())
                .documentLink(DocumentLink.builder()
                                  .documentUrl(document.getValue().getDocumentUrl())
                                  .documentBinaryUrl(document.getValue().getDocumentBinaryUrl())
                                  .documentFilename(document.getValue().getDocumentFileName()).build())
                .build()).collect(Collectors.toList());

        List<Element<BundlingRequestDocument>> particulars = ElementUtils.wrapElements(bundlingServedDocFiles);
        return ServedDocument.builder().particularsOfClaimDocument(particulars).build();
    }
```

Whether one sees any benefit in it or not is subjective and streams can get complicated. Still, short-circuiting the "if" may improve readability.

Also consider that you're not fully covering all the branches of the "if" so I will want a test about the missing branch.